src/intrprtr.c

```diff
@@ -584,101 +530,99 @@ void            IntrIfBegin ( void )
     // be executed, either because a previous branch is always executed
     // (i.e., it has a 'true' condition), or else because the current branch
     // has a 'false' condition
-    if ( STATE(IntrIgnoring)  > 0 ) { STATE(IntrIgnoring)++; return; }
-    if ( STATE(IntrCoding)    > 0 ) { CodeIfBegin(); return; }
+    if ( intr->ignoring  > 0 ) { intr->ignoring++; return; }
+    if ( intr->coding    > 0 ) { CodeIfBegin(); return; }
 
 }
 
-void            IntrIfElif ( void )
+void IntrIfElif(IntrState * intr)
 {
     /* ignore or code                                                      */
     SKIP_IF_RETURNING();
     SKIP_IF_IGNORING();
-    if ( STATE(IntrCoding)    > 0 ) { CodeIfElif(); return; }
+    if ( intr->coding    > 0 ) { CodeIfElif(); return; }
 
 }
 
-void            IntrIfElse ( void )
+void IntrIfElse(IntrState * intr)
 {
     /* ignore or code                                                      */
     SKIP_IF_RETURNING();
     SKIP_IF_IGNORING();
-    if ( STATE(IntrCoding)    > 0 ) { CodeIfElse(); return; }
+    if ( intr->coding    > 0 ) { CodeIfElse(); return; }
 
 
     /* push 'true' (to execute body of else-branch)                        */
-    PushObj( True );
+    PushObj(intr, True);
```

Comment:
Another note: almost all changes which add `IntrState * intr` as a parameter, resp. pass `intr` as an argument, could be removed again if we turned `IntrState` into a simple C++ class, and turned `IntrIfElse` etc. into member functions of that.